lib/api/core/models/repositories/repository.js

```diff
@@ -230,14 +230,14 @@ Repository.prototype.loadFromCache = function (id, opts) {
 Repository.prototype.load = function (id, opts) {
   var deferred = q.defer();
 
-  if (this.cacheEngine === null) {
+  if (!this.cacheEngine) {
     return this.loadOneFromDatabase(id);
   }
 
   this.loadFromCache(id, opts)
     .then(function (object) {
       if (object === null) {
-        if (this.readEngine === null) {
+        if (!Boolean(this.readEngine)) {
```

Comment:
I don't get the condition. To me, the readEngine should either be set to null or to an readEngine service instance.
If the readEngine is an instance of ES service, my understanding here is that the object won't ever be fetched if not found in the cache.